lib/index.js

```diff
@@ -123,35 +133,80 @@ function createRouter(spec, options) {
     }
 
     const routes = spec.resolved.paths;
-    _.forOwn(routes, function (methods, route) {
+    _.forOwn(routes, function(methods, route) {
+      const originalRoute = route;
 
-      /* Convert from spec to router parameter syntax: */
-      route = route.replace(/{[^/]*}/g, function replacer(match) {
-        /* Cut first and last character, prepend with colon: */
-        return ':' + match.slice(1, -1);
-      });
+      route = formatRoute(route);
 
-      _.forOwn(methods, function (methodInfo, method) {
+      _.forOwn(methods, function(methodInfo, method) {
         const controllerName = methodInfo['x-controller'];
         const controllerMethodName = methodInfo['x-controller-method'];
 
         const controllerMethod = getControllerMethod(controllerDirectory, controllerName, controllerMethodName, strictMode, method, route);
 
         if (router[method]) {
           checkMethodInfo(method, route, methodInfo);
-
           const requestValidator = createRequestValidator(spec, method, route, methodInfo.parameters);
-          router[method](route, requestValidator, controllerMethod);
+          const middlewarePluginList = spec.resolved['middleware-plugins'];
```

Comment:
Might have to have `x-` prefixed in order to comply with open api spec schema definition.